// @nbcon/enterprise-sdk/api.ts
import { createClient, SupabaseClient } from "@supabase/supabase-js";

function getEnvVar(key: string): string | undefined {
  // Next.js uses process.env for environment variables
  // Support both NEXT_PUBLIC_ prefix (client-side) and direct access (server-side)
  const value =
    (typeof process !== "undefined" && process.env?.[key]) ||
    (typeof process !== "undefined" && process.env?.[`NEXT_PUBLIC_${key}`]) ||
    (typeof window !== "undefined" && (window as any).__ENV__?.[key]);

  return value;
}

// Lazy-load supabase client to avoid errors when env vars are not set
let supabaseClient: SupabaseClient | null = null;

function getSupabaseClient(): SupabaseClient {
  if (!supabaseClient) {
    const url = getEnvVar("SUPABASE_URL") || getEnvVar("NEXT_PUBLIC_SUPABASE_URL");
    const key = getEnvVar("SUPABASE_SERVICE_ROLE_KEY") || getEnvVar("NEXT_PUBLIC_SUPABASE_ANON_KEY");

    if (!url || !key) {
      // Return a mock client that will fail gracefully
      // This allows the module to load even if env vars are missing
      console.warn("Supabase environment variables not set. Some features may not work.");
      // Create a client with placeholder values - it will fail on actual use
      supabaseClient = createClient("https://placeholder.supabase.co", "placeholder-key");
    } else {
      supabaseClient = createClient(url, key);
    }
  }
  return supabaseClient;
}

export const supabase: SupabaseClient = new Proxy({} as SupabaseClient, {
  get(_target, prop) {
    return getSupabaseClient()[prop as keyof SupabaseClient];
  },
}) as SupabaseClient;

export async function fetchData<T = any>(table: string): Promise<T[]> {
  const { data, error } = await supabase.from(table).select("*");
  if (error) throw new Error(error.message);
  return data || [];
}

export async function insertData<T = any>(
  table: string,
  payload: Partial<T>
): Promise<T> {
  const { data, error } = await supabase.from(table).insert(payload).select().single();
  if (error) throw new Error(error.message);
  return data;
}

export async function updateData<T = any>(
  table: string,
  id: string,
  payload: Partial<T>
): Promise<T> {
  const { data, error } = await supabase
    .from(table)
    .update(payload)
    .eq("id", id)
    .select()
    .single();
  if (error) throw new Error(error.message);
  return data;
}
