// @nbcon/enterprise-sdk/integrations/stripe.ts
import Stripe from "stripe";

function getStripeSecret(): string | undefined {
  // STRIPE_SECRET_KEY should NEVER be exposed to client-side code
  // Only access it server-side
  if (typeof window !== "undefined") {
    // Client-side: return undefined (Stripe operations should be server-side only)
    return undefined;
  }

  // Server-side: check process.env
  return (
    (typeof process !== "undefined" && process.env?.STRIPE_SECRET_KEY) ||
    undefined
  );
}

// Lazy-load Stripe client to avoid errors when env vars are not set or in client-side code
let stripeClient: Stripe | null = null;

function getStripeClient(): Stripe | null {
  // Only create Stripe client server-side
  if (typeof window !== "undefined") {
    return null;
  }

  if (!stripeClient) {
    const secret = getStripeSecret();

    if (!secret) {
      // Return null instead of throwing - allows module to load
      console.warn("STRIPE_SECRET_KEY not set. Stripe operations will not work.");
      return null;
    }

    stripeClient = new Stripe(secret, {
      apiVersion: "2023-10-16" as any, // Using older Stripe version (14.x) - will be updated in remediation
    });
  }

  return stripeClient;
}

// Export a proxy that lazily loads the Stripe client
export const stripe: Stripe | null = new Proxy({} as Stripe, {
  get(_target, prop) {
    const client = getStripeClient();
    if (!client) {
      // Return a no-op function for client-side or when env vars are missing
      if (typeof prop === 'string' && typeof ({} as any)[prop] === 'function') {
        return () => {
          throw new Error("Stripe client not available. STRIPE_SECRET_KEY must be set server-side.");
        };
      }
      return null;
    }
    return (client as any)[prop];
  },
}) as Stripe | null;

export async function createSubscription(customerId: string, priceId: string) {
  const client = getStripeClient();
  if (!client) {
    throw new Error("Stripe client not available. STRIPE_SECRET_KEY must be set server-side.");
  }
  return await client.subscriptions.create({
    customer: customerId,
    items: [{ price: priceId }],
  });
}

export async function cancelSubscription(subscriptionId: string) {
  const client = getStripeClient();
  if (!client) {
    throw new Error("Stripe client not available. STRIPE_SECRET_KEY must be set server-side.");
  }
  return await client.subscriptions.cancel(subscriptionId);
}
