// @nbcon/.ignored_ai-core/agentRegistry.ts
import { AgentConfig } from './interfaces';

export const agentRegistry: Record<string, AgentConfig> = {
  civil: {
    id: "civilAgent",
    description: "Handles site design, grading, and material estimation.",
    model: "gpt-5",
    context: "civil-engineering",
    maxTokens: 4000,
    temperature: 0.3,
  },
  electrical: {
    id: "electricalAgent",
    description: "Generates load schedules, panel design, and wiring plans.",
    model: "gpt-5",
    context: "electrical-engineering",
    maxTokens: 4000,
    temperature: 0.3,
  },
  mechanical: {
    id: "mechanicalAgent",
    description: "Performs HVAC load calculations and piping analysis.",
    model: "gpt-5",
    context: "mechanical-engineering",
    maxTokens: 4000,
    temperature: 0.3,
  },
  survey: {
    id: "surveyAgent",
    description: "Processes GNSS, LiDAR, and topographic datasets.",
    model: "gpt-5",
    context: "survey-engineering",
    maxTokens: 4000,
    temperature: 0.3,
  },
  gis: {
    id: "gisAgent",
    description: "Analyzes spatial data and creates mapping outputs.",
    model: "gpt-5",
    context: "geospatial-analysis",
    maxTokens: 4000,
    temperature: 0.3,
  },
  geotechnical: {
    id: "geotechAgent",
    description: "Analyzes soil and foundation design data.",
    model: "gpt-5",
    context: "geotechnical-engineering",
    maxTokens: 4000,
    temperature: 0.3,
  },
  environmental: {
    id: "environmentalAgent",
    description: "Handles environmental impact assessments, compliance, and remediation planning.",
    model: "gpt-5",
    context: "environmental-engineering",
    maxTokens: 4000,
    temperature: 0.3,
  },
};
